### ISSUE TITLE
Issue recognize_commands micro_speech demo

### ISSUE THREAD

**Issue by RobotEdh** [tensorflow/tensorflow#28516]
**System information**
run on SparkfunEdge Board Apollo3 operating in Burst Mode (96MHz)

Bin build on the following linux env:
Linux edh-VirtualBox 4.18.0-17-generic #18.04.1-Ubuntu SMP Fri Mar 15 15:27:12 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
TensorFlow installed from (source or binary): Git
TensorFlow version: 1.13.1
Python version:Python 2.7.15rc1
Installed using virtualenv? pip? conda?: pip
Bazel version (if compiling from source): No
GCC/Compiler version (if compiling from source): gcc version 7.3.0 (Ubuntu 7.3.0-27ubuntu1~18.04)
CUDA/cuDNN version: No
GPU model and memory: No

**Description**
The micro_speech example does not work.
The 'Yes' or 'No' are not regognized.
It seems a logic issue in the source 

> tensorflow/lite/experimental/micro/examples/micro_speech/recognize_commands.cc

The TF model returns a good score > 200 following call to` interpreter.Invoke() `and the vector output reurned by `interpreter.output(0)` is most of the time correct in the source 

> main.cc

 **BUT there is a average computation**  that breaks the results in the source 

> recognize_commands.cc

With the default configuration (`average_window_duration_ms = 1000`) , the average is computed on about 10 model outputs. 
The issue is that when I say 'Yes', only 1or 2 model ouputs detect 'yes with a score > 200. So the average computed on 10 outputs provides a average score always smaller than the threshold equal to 200.
I have tried to reduce the param `average_window_duration_ms = 300`, then the average is computed on 3 or 4 outputs but the result is still < 200.
So to solve the issue, **I have removed the average and then it works**.

I don't understand the purpose of the average of the outputs. When I say 'yes', the model detects only one occurence of 'yes', There is no stuttering in the input data?

Could you please confirm my analyse and my fix.

Thanks in advance

**Comment by RobotEdh** [comment]
Hello, no update since one month.
lite/micro speech project is aborted ?
what means "awaiting tensorflower" ?

Happy to contribute if i can help

**Comment by petewarden** [comment]
The averaging is used to try to balance the sensitivity and robustness of the speech detection. If no averaging is used (or the average window is very short), then it's more likely to detect spoken words, but also more likely to make mistakes. It will depend a lot on your application whether you care more about false positives or false negatives.

We are also aware of the broader issue that the overall accuracy of the yes/no model isn't as high as we'd like. We believe a lot of improvement can be had by gathering a larger training data set, and one way to help is by contributing your own voice here: https://aiyprojects.withgoogle.com/open_speech_recording

**Comment by petewarden** [comment]
Closing this for now, since there's been no updates for a while.
